internal/storage/memory.go

```diff
@@ -76,76 +153,98 @@ func (s *memoryIssuerService) Create(ctx context.Context, iss types.Issuer) (*ty
 	return &iss, nil
 }
 
-func (s *memoryIssuerService) GetByID(ctx context.Context, id string) (*types.Issuer, error) {
-	row := s.db.QueryRow(`SELECT id, name, uri, jwksuri, mappings FROM issuers WHERE id = $1;`, id)
+// GetIssuerByID gets an issuer by ID.
+func (s *memoryIssuerService) GetIssuerByID(ctx context.Context, id string) (*types.Issuer, error) {
+	query := fmt.Sprintf("SELECT %s FROM issuers WHERE id = $1", issuerColumnsStr)
+	row := s.db.QueryRow(query, id)
 
-	iss, err := s.scanIssuer(row)
+	return s.scanIssuer(row)
+}
 
-	switch {
-	case errors.Is(err, sql.ErrNoRows):
-		err := v1.ErrorIssuerNotFound{
-			Label: id,
-		}
+// GetByURI looks up the given issuer by URI, returning the issuer if one exists.
+func (s *memoryIssuerService) GetIssuerByURI(ctx context.Context, uri string) (*types.Issuer, error) {
+	query := fmt.Sprintf("SELECT %s FROM issuers WHERE uri = $1", issuerColumnsStr)
```

Comment:
With the addition of `TenantID`, this isn't guaranteed to return the correct issuer row.